src/components/RegisterForm.vue

```diff
@@ -0,0 +1,190 @@
+<template>
+  <!-- v-clock will hide everything from rendering until the whole app is ready -->
+  <div class="form" v-cloak>
+    <h1>{{msg}}</h1>
+    <!-- horizintal line in the form -->
+    <hr size="3" noshade><br>
+
+    <h2>Full Name:</h2>
+    <div v-if="emptyFname">*please enter your full name*<br><br><input v-model="fname"></div>
+    <div v-else-if="fnameValidation" > <input v-model="fname" class="Valid"></div>
+    <div v-else><p>*name is not valid*</p><input v-model="fname" class="NoValid"></div>
+
+    <h2>Mail:</h2>
+    <div v-if="emptyMail">*please enter your email*<br><br><input v-model="mail"></div>
+    <div v-else-if="emailValidation" > <input v-model="mail" class="Valid"></div>
+    <div v-else><p>*email is not valid*</p><input v-model="mail" class="NoValid"></div>
+
+    <h2>Phone Number:</h2>
+    <div v-if="emptyPhoneNumber">*please enter phone number*<br><br><input v-model="phonenum"></div>
+    <div v-else-if="phonenumValidation"> <input v-model="phonenum" class="Valid"></div>
+    <div v-else><p>*number is not valid*</p><input v-model="phonenum" class="NoValid"></div>
+    
+    <h2>Gender:</h2>
+    <div v-if="genderValidation">
+    <select v-model="gender" class="Valid">
+      <option disabled value=""></option>
+      <option value="Male">Male</option>
+      <option value="Female">Female</option>
+      <option value="Both">Both</option>
+    </select></div>
+    <div v-else>*gender has to be selected*<br><br>
+    <select v-model="gender">
+      <option disabled value=""></option>
+      <option value="Male">Male</option>
+      <option value="Female">Female</option>
+      <option value="Both">Both</option>
+    </select></div>
+
+    <h2>Your Avatar:</h2>
+    <label v-if="imgNoValidation">*please select an Avatar*<br><br></label>
+    <label v-for="img in imgs" :key="img.src">
+      <input type="radio" v-model="image" :value="img.value">
+      <img :src="img.src" alt="avatar" /></label><br><br>
```

Comment:
your alt is the same for all options - is it on purpose? makes much more sense IMO to have the name of the specific avatar